### DISCUSSION TITLE
WhatsApp integration

### DISCUSSION THREAD

**Discussion by jeovajireh** [chatwoot/chatwoot#1638] [Feature Requests]
Topic for Pranav Raj S.

Hello Pranav, I noticed that the colleague closed the topic as much as he desperately asked to be deleted, this is probably due to the fact that he wants to use chatwoot as a service. This is our case, but found injustice finalize the topic as well, so I opened it.

We are a small company in Brazil, I am interested in discussing this integration with you, I believe we can help each other. As I said, we already have 60% of the code ready and the remaining 40% that I myself have documented and delivered to our programmer, only we decided to suspend the financial investment in the project because the current form of Chatwoot is very different from the others. We made this decision because we came to the conclusion that with 50% of the amount we still need to invest and half the time we managed to integrate another chat.

If you want to talk directly, my email is camalau@gmail.com. Unfortunately I can't call you because I don't have full command of English, having to use translation tools to communicate.

**Comment by jeovajireh** [comment]
@pranavrajs 

Do you think it is possible to simplify the integration in the Chatwoot part?

**Reply by pranavrajs** [reply]
@viapratica Are you considering unofficial Whatsapp APIs?

**Reply by pranavrajs** [reply]
Can you please share which API provider are you trying to use?

**Reply by jeovajireh** [reply]
An own API. The great difficulty for any API is that Chatwoot has some complicated requirements.
The practical way would be to get the whatsapp number that is passed as a contact and the arrival of the message as the beginning of a conversation. As for the inbox we could easily add to the webhook to start the conversation with the correct inbox.

If you can make a post with the phone and message fields become a new contact and new conversation everything will be very simple. Maybe you could create a copy of the current API, make the changes and put it as a new channel, API 2.0.

**Reply by jeovajireh** [reply]
One point to be noted, the API's operation that I did the test is similar to any other unofficial one, it will also be in the same line as the official WhatsApp partners, that is, it made it possible for one to be easier for everyone, of course this is it's just the beginning of what needs to be done. Do you think of using this in the free version I would be marketed?

**Reply by sojan-official** [reply]
@viapratica  Our API is primarily optimized for the web messenger, where a user could create multiple conversations across multiple browser sessions. To support that use case we will have to have additional data modelling like contact_inboxes etc.  The entire product is built on top of that data modelling. 


you could refer the documentation. for how to work with the APIs
https://www.chatwoot.com/docs/channels/api/send-messages 

if you can't work with the current structure of APIs, another option you could take is forking the repo and add additional APIs to the fork as required by your integration.

**Reply by jeovajireh** [reply]
> @viapratica Our API is primarily optimized for the web messenger, where a user could create multiple conversations across multiple browser sessions. To support that use case we will have to have additional data modelling like contact_inboxes etc. The entire product is built on top of that data modelling.
> 
> you could refer the documentation. for how to work with the APIs
> https://www.chatwoot.com/docs/channels/api/send-messages
> 
> if you can't work with the current structure of APIs, another option you could take is forking the repo and add additional APIs to the fork as required by your integration.

Hi @sojan-official! 

I'm glad for your participation. So, as I said in another post, we already have everything documented for integration with our API. I started this discussion after a user opened a topic and shortly after closing it, he requested that it be deleted with the intention of retaining knowledge. The suggestions I made would be a way to facilitate the integration, because today you need to do some interactions with the API to be able to start a conversation.

**Reply by E-miles** [reply]
I am not sure because I have done this before, So I just say unofficial API's maybe WhatsApp doesn't allow them to integrate with it... but that API's can be used with modified version of WhatsApp like [gb whatsapp](https://gbapps.net/gbwhatsapp-apk/) as I said before I don't know it's work or not and I also don't recommend you to use it because it is a third party modified application...
